### PR TITLE
Add more connection executor tests (native types, select data, cast for output, etc.)

### PR DESCRIPTION
**Pull request by KonstantAnxiety**

### CODE REVIEW

File: lib/dl_api_lib_testing/dl_api_lib_testing/connector/complex_queries.py [hunk 1]
```diff
@@ -88,6 +89,7 @@ def test_null_dimensions(self, control_api, data_api, db, saved_connection_id):
             C("sales", UserDataType.integer, vg=lambda rn, **kwargs: raw_data[rn]["sales"]),
         ]
         db_table = make_table(db, columns=columns, rows=len(raw_data))
+        request.addfinalizer(functools.partial(db.drop_table, db_table.table))
```
[the remark this answers is not in the document]
Comment (KonstantAnxiety) [reply lib/dl_api_lib_testing/dl_api_lib_testing/connector/complex_queries.py:92]: And make a fixture for each test that requires a custom table? Is that what you're suggesting?

[the remark this answers is not in the document]
Comment (KonstantAnxiety) [reply lib/dl_api_lib_testing/dl_api_lib_testing/connector/complex_queries.py:92]: I thought about adding a context manager implementation of make_table, but I don't think that this huge additional indentation is better than a finalizer

[the remark this answers is not in the document]
Comment (MCPN) [reply lib/dl_api_lib_testing/dl_api_lib_testing/connector/complex_queries.py:92]: hm, yeah, i've probably mixed up fixtures and context managers. finalizers smell a little bit of golang, but for tests i think they're fine)

File: lib/dl_connector_postgresql/dl_connector_postgresql_tests/db/core/test_connection_executor.py [hunk 2]
```diff
@@ -97,6 +63,42 @@ class TestPostgreSQLSyncConnectionExecutor(
...
+        }
+
+    def test_type_recognition(
+        self,
+        request,
+        db: Db,
+        sync_connection_executor: SyncConnExecutorBase,
+        enabled_citext_extension,
+    ) -> None:
+        super().test_type_recognition(request, db, sync_connection_executor)
```
[the remark this answers is not in the document]
Comment (MCPN) [reply lib/dl_connector_postgresql/dl_connector_postgresql_tests/db/core/test_connection_executor.py:100]: and now you can decorate the whole class and remove the redefinition)

**Review by MCPN — CHANGES_REQUESTED** [review]

**Review by MCPN — APPROVED** [review]

**Pull request merged**